app/geo/gui/src/ts/twxplore/gui/geo/attributeStrategies/attributeStrategies.ts

```diff
@@ -1,82 +1,156 @@
 // In an OOP Language -
+
+import { FeatureAttributeName } from "../states/map/FeatureAttributeName";
+
 // TypeScript
-interface FeatureAttributeFunctions {
+interface FeatureAttributeTypes {
     isNumeric() : boolean
     isString(): boolean
+    getFilterType(): string
+    ignore(): boolean
+    attributeName: FeatureAttributeName
 }
 
-enum Attributes {
-    LABEL = "label",
-    TYPE = "type",
-    URI = "uri",
-    FREQUENCY = "frequency",
-    TIMESTAMP = "timestamp",
-    LOCALITY = "locality",
-    POSTALCODE = "postalcode",
-    REGIONS =  "regions",
-    TRANSMISSIONPOWER = "transmissionPower",
-    GEOMETRY = "geometry",
-}
 
-export class FeatureAttribute implements FeatureAttributeFunctions{
-    private attributeName: string 
-    private subclass: any
-    constructor(attributeName: string) {
+ export abstract class FeatureAttribute implements FeatureAttributeTypes{
+    public attributeName: FeatureAttributeName 
+    constructor(attributeName: FeatureAttributeName) {
```

Comment:
Just make name() another abstract method. That way subclasses don't need a constructor. (And you can eventually transition to using interfaces.)

Reply:
Also, you can drop prefixes when the context is clear.

For example, you're going to be doing featureAttribute. , so it can be featureAttribute.name() (or rather .name, you can have abstract getters `abstract get name(): string`. Rather than featureAttribute.attributeName, which is repetitive.